internal/pkg/service/buffer/definition/repository/branch_repo.go

```diff
@@ -0,0 +1,172 @@
+package repository
+
+import (
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/keboola"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/definition/key"
+	serviceError "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+const (
+	MaxBranchesPerProject = 100
+)
+
+type BranchRepository struct {
+	telemetry telemetry.Telemetry
+	clock     clock.Clock
+	client    etcd.KV
+	schema    branchSchema
+	all       *Repository
+}
+
+func newBranchRepository(d dependencies, all *Repository) *BranchRepository {
+	return &BranchRepository{
+		telemetry: d.Telemetry(),
+		clock:     d.Clock(),
+		client:    d.EtcdClient(),
+		schema:    newBranchSchema(d.EtcdSerde()),
+		all:       all,
+	}
+}
+
+func (r *BranchRepository) List(parentKey keboola.ProjectID) iterator.DefinitionT[definition.Branch] {
+	return r.list(r.schema.Active(), parentKey)
```

Comment:
V repositories sa pouziva vylepsena verzia `etcdop` frameworku, z predchadzajuceho [PR](https://github.com/keboola/keboola-as-code/pull/1441).

Vsetky operacie su atomicke (zalozene na etcd txn).

Reply:
Kod repository aj testy vsetkych 3 entity: `Branch`,`Source`,`Sink` je prakticky rovnaky.

Dala by sa z toho spravit sablona a repository aj testy generovat cez `go generate`, ale kedze mam iba 3 entity, a nevyzera to, ze ich bude casom nejak viac, tak to nebudem teraz prehanat.